lib/sqlalchemy/dialects/postgresql/asyncpg.py

```diff
@@ -1060,10 +1086,16 @@ def do_terminate(self, dbapi_connection) -> None:
 
     def create_connect_args(self, url):
         opts = url.translate_connect_args(username="user")
+        multihosts, multiports = self._split_multihost_from_url(url)
 
         opts.update(url.query)
+
+        if multihosts:
+            opts["host"] = multihosts
+            opts["port"] = multiports
```

Comment:
Here, I suggest convert `host` and `port` from tuples to lists.
asyncpg is waiting for a list.

Reply:
Also, asyncpg doesn't accept None values (And unfortunately don't use default values in case of None)
So, hosts and ports in a list should match, and shouldn't be None values

Reply:
yeah what should we do , raise an error?  I would rather not guess a default

Reply:
we would have this raise be on the asyncpg side; asyncpg gets the list of ports, does "if None in ports: raise ..."

Reply:
The way I see it, the best way to organize this is to conform asyncpg library as much as possible.
And as far as library don't want to get empty values, we should raise an exception.

Imagine situation some engineer didn't set configuration correctly (For example number of ports and host don't match). And if this exception is risen from asyncpg side, it will look something like this:
```
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
```
Which is absolutely unreadable and very hard to understand what is going on, and where is it happened.

So, I suggest filter all the empty values from `multihosts` and `multiports`
, and check if their length match. (also we can check if each port is a string representing an integer)
Otherwise raise an `ArgumentError` exception with intelligible message.

Reply:
Something like this 

```python
if not all(multihosts):
    raise exc.ArgumentError(
        "Some hosts is not specified"
        " in a connection string"
    )
if not all(multiports) or not all(
        map(
            lambda x: isinstance(x, str) and x.isdigit(),
            multiports
        )
):
    raise exc.ArgumentError(
        "Some ports is not specified"
        " in a connection string"
    )
    
```

Reply:
OK give it a try.  add a new test for it in postgresql/test_dialect.py, using just asyncpg.dialect() the way the other tests do.  the tests for this feature shouldn't require that the underlying `asyncpg` library is actually imported (that is, dont use `create_engine()` in the test).

Reply:
@zzzeek done, have a look